services/discordService.js

```diff
@@ -52,7 +52,31 @@ const setInDiscordFalseScript = async () => {
   await Promise.all(updateUsersPromises);
 };
 
+const generateAuthTokenForCloudflare = async () => {
+  const expiry = config.get("rdsServerlessBot.ttl");
+  const privateKey = config.get("rdsServerlessBot.rdsServerLessPrivateKey");
+  const authToken = await jwt.sign({}, privateKey, {
+    algorithm: "RS256",
+    expiresIn: expiry,
+  });
+  return authToken;
+};
+
+const addRoleToUser = async (userid, roleid) => {
+  const authToken = await generateAuthTokenForCloudflare();
+  const response = await (
+    await fetch(`${DISCORD_BASE_URL}/roles/add`, {
+      method: "PUT",
+      headers: { "Content-Type": "application/json", Authorization: `Bearer ${authToken}` },
+      body: JSON.stringify({ userid, roleid }),
+    })
+  ).json();
```

Comment:
- I think a better way would be to handle this in two separate variables and add a try-catch block to the function. What do you say ?

Reply:
The consumer of this service has a try catch block that would handle it, Wont that suffice? would love to hear your inputs on this.